### PR TITLE
Ddr base

### PR DESCRIPTION
**Pull request by jordanagreen**
WIP - pretty much everything other than search page and minor cosmetic stuff is done, DDR scraper/importer could use improvement but they work fine for now.

### CODE REVIEW

**Comment by jordanagreen** [comment]
Search page works now.

File: statistik/controller.py [hunk 1]
```diff
@@ -526,14 +579,19 @@ def elo_rate_charts(chart1_id, chart2_id, user, draw=False, rate_type=0):
...
     """
     Get songs ranked by Elo ranking, formatted for template usage
-    :param int level:       Level to sort by (1-12)
+    :param int game:        The game to get songs from (0-1)
+    :param int level:       Level of songs to sort by (1-12) for IIDX, (1-19) for DDR
     :param str rate_type:   Rating type (refer to Chart model for options)
     :rtype list:            List of dicts containing chart/ranking data
     """
-    matched_charts = Chart.objects.filter(difficulty=int(level), type__lt=3) \
+    # TODO: just add a game field to song model at some point so this can be avoided
```
Comment (benhgreen) [inline statistik/controller.py]: remove pls

File: statistik/controller.py [hunk 2]
```diff
@@ -526,14 +579,19 @@ def elo_rate_charts(chart1_id, chart2_id, user, draw=False, rate_type=0):
...
+    versions = [str(v[0]) for v in VERSION_CHOICES[game]]
+    # singles difficulties only
+    sng = {IIDX: [str(i) for i in range(0, 3)], DDR: [str(i) for i in range(100, 105)]}
```
Comment (benhgreen) [inline statistik/controller.py]: maybe worth storing the ranges in constants with the games?

Comment (benhgreen) [inline statistik/controller.py]: also don't abbreviate to `sng` pls v confusing

File: statistik/controller.py [hunk 3]
```diff
@@ -597,40 +659,53 @@ def make_nav_links(level=None, style='SP', version=None, user=None, elo=None,
     :param int clear_type:  Rating type (refer to Chart model for options)
     :rtype list:            List of tuples of format (link text, link)
     """
-    ret = [(_('INDEX'), reverse('index')),
-           (_('SEARCH'), reverse('search'))]
+    ret = [(_('INDEX'), reverse('index', kwargs={'game': GAME_CHOICES[game][1]})),
+           (_('SEARCH'), reverse('search', kwargs={'game': GAME_CHOICES[game][1]}))]
```
Comment (benhgreen) [inline statistik/controller.py]: pls at least set `GAME_CHOICES[game][1]` to a name and use it throughout method, this applies elsewhere in this PR

File: statistik/forms.py [hunk 4]
```diff
@@ -30,14 +35,21 @@ def is_valid(self):
             self.add_error('reenter_password', _('Passwords do not match.'))
             return False
 
-        if len(data.get('best_techniques')) > 3:
-            self.add_error('best_techniques', _('Please select no more than 3.'))
+        if len(data.get('best_techniques_iidx')) > 3:
```
Comment (benhgreen) [inline statistik/forms.py]: pls remove repeated code

File: statistik/views.py [hunk 5]
```diff
@@ -149,15 +163,24 @@ def chart_view(request):
 
     # get reviews for this chart, cache users for username and playside lookup
     context['reviews'] = get_reviews_for_chart(chart_id)
+    if chart.song.game == IIDX:
+        style = chart.get_type_display()[:2]
+    else:
+        style = chart.get_type_display()[1:]
+        if style[0] == 'E':
```
Comment (benhgreen) [inline statistik/views.py]: comment

**Pull request merged**